app/models/ext_management_system.rb

```diff
@@ -586,6 +598,7 @@ def stop_event_monitor_queue_on_credential_change
     if event_monitor_class && !self.new_record? && self.credentials_changed?
       _log.info("EMS: [#{name}], Credentials have changed, stopping Event Monitor.  It will be restarted by the WorkerMonitor.")
       stop_event_monitor_queue
+      network_manager.stop_event_monitor_queue if respond_to?(:network_manager) && network_manager
```

Comment:
This feel misplaced all the way up in `ExtManagementSystem`.  I see the same code down in `Openstack::ManagerMixin`.  Maybe a middle ground would be to have this in `CloudManager`?